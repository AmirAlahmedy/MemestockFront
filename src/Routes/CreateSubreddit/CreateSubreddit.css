.createSubredditContainer{
    font-style: arial;
    text-align: center;
    color:rgb(132, 134, 136);
    width:80%;
    overflow: hidden;
    margin-top:25px;
    margin-bottom:15px; 
    background-color: rgb(26, 26, 27);
    border-bottom: 1px solid rgb(52, 53, 54);
    border-right:  1px solid rgb(52, 53, 54);
    border-left:  1px solid rgb(52, 53, 54);
    border-top:  1px solid rgb(52, 53, 54);
    border-radius: 6px;
}
h3{
    font-style: inherit;
    font-size:20px;
    font-weight: 500;
    float:center;       
    margin:0;
    padding-top:10px;
    padding-left:1%;
    border:0;
}

.formGroupSubredditComponent{
    font-size:18px;
    text-align: left;
    padding-left:20px;
    padding-right:20px;
    padding-bottom:15px;
    color:rgb(132, 134, 136);
}
.formGroupSubredditComponent label{
    color:rgb(116, 193, 219);
}
.formGroupSubredditComponent textarea{
    margin-top:10px;
    margin-bottom:10px;
    border-radius:10px;
    width:100%;
}
.formGroupSubredditComponent input{
    margin-top:10px;
    margin-bottom:10px;
    border-radius:10px;
    width:100%;
}
.formGroupSubredditComponent label {
    float:left;
}
.srSubredditPageCreateButton {
    margin:10px 30px 30px;
    padding:10px;
    width:50%;
    border-radius: 7px;
    background-color:lightgrey;
    font-size:18px;
    font-weight: 600;
    cursor:pointer;
}
.srSubredditPageCreateButton:hover{
    color:rgb(233, 172, 92);
    background-color:rgb(26, 26, 27);
}