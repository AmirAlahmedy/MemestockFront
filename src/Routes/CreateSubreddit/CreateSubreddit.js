import React, { Component } from 'react';
import './CreateSubreddit.css';
import axios from '../../axios-orders';

class CreateSubreddit extends Component {

  constructor(props) {
  super(props);
  this.state = {
    Subreddit: '',
    Rule1:'',
    Rule2 :'',
    Rule3 : '',
    bio:''
  }
  
    this.handleChange = this.handleChange.bind(this); 
    this.handleSubmit = this.handleSubmit.bind(this);
};



handleChange(event) {
this.setState({ [event.target.name]: event.target.value });
}
  /**
     * For sending a creation of subreddit request to the backend 
     * @function handleSubmit
     * @param {event} - onClick event 
     */

handleSubmit (e){
  e.preventDefault();

  let checker ="";
  if (document.getElementById("srSubredditName").value===checker)
  {
    alert ("Please provide a Subreddit name!");
    return ;
  }
  else if (document.getElementById("srSubredditRule1").value===checker)
  { alert ("Please provide a Subreddit Rule!");
    return ;
  }
  else if (document.getElementById("srSubredditRule2").value===checker)
  { alert ("Please provide a Subreddit Rule!");
    return ;
  }
  else if (document.getElementById("srSubredditRule3").value===checker)
  { alert ("Please provide a Subreddit Rule!");
    return ;
  }
  else if (document.getElementById("srSubredditBio").value===checker)
  { alert ("Please provide a Subreddit Bio For the audience!");
    return ;
  }
  var srdata ={
    srName : document.getElementById("srSubredditName").value,
    srRules: [document.getElementById("srSubredditRule1").value,document.getElementById("srSubredditRule2").value,document.getElementById("srSubredditRule3").value]
    } 
  let headers = {
        auth: localStorage.getItem("token") 
  } 
    axios.post( 'sr/create',srdata,{"headers": headers})
    .then(res => {
      console.log(res);
      console.log(res.status);
      window.location.href = "/r/" + srdata.srName;
    })
    .catch(error => {
    console.log("Axios Error: ",error.response)

  });
}

componentDidMount() {
  document.body.classList.add("verticalAlign");
}
componentWillUnmount() {
  document.body.classList.remove("verticalAlign");
}
  render() {
    return (

      <div className="createSubredditContainer">
      <h3>CREATE A SUBREDDIT</h3>
      <hr></hr>
      <form onSubmit={this.handleSubmit}>
        <div className="formGroupSubredditComponent">
        <label for="SubredditName">Enter Subreddit Name</label>
        <br></br>
        <input type="text" name="text" id="srSubredditName" placeholder = "Think of something remarkable!" onChange={this.handleChange} value={this.state.value}/>   
        </div>
        <div className="formGroupSubredditComponent">
        <label for="SubredditRule">Subreddit Rule:</label>
        <textarea type="textarea" name="text" id="srSubredditRule1" placeholder = "ex: Flamers will be banned! " onChange={this.handleChange} value={this.state.value}/>   
        </div>
        <div className="formGroupSubredditComponent">
        <label for="SubredditRule">Subreddit Rule:</label>
        <textarea type="textarea" name="text" id="srSubredditRule2" placeholder = "ex: Flamers will be banned! " onChange={this.handleChange} value={this.state.value}/>  
        </div>
        <div className="formGroupSubredditComponent">
        <label for="SubredditRule">Subreddit Rule:</label>
        <textarea type="textarea" name="text" id="srSubredditRule3" placeholder = "ex: Flamers will be banned! " onChange={this.handleChange} value={this.state.value}/>  
        </div>
        <div className="formGroupSubredditComponent">
        <label for="SubredditBio">Subreddit Bio:</label>
        <textarea type="textarea" name="text" id="srSubredditBio" placeholder = "ex: This subreddit is for people who..etc. " onChange={this.handleChange} value={this.state.value}/>  
        </div>
        <button className="srSubredditPageCreateButton">CREATE</button>  
      </form>
      </div>
   
    )
  }
}
export default CreateSubreddit;
