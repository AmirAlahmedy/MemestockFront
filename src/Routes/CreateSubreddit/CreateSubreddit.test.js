import React from "react";
import { create } from "react-test-renderer";
import CreateSubreddit from "./CreateSubreddit";

describe("CreateSubreddit component snapshot", () => {
    test("it matches the snapshot", () => {
      const component = create(<CreateSubreddit />);
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
  
