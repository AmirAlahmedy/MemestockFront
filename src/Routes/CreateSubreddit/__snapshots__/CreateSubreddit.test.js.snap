// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePost component snapshot it matches the snapshot 1`] = `
<div
  className="createSubredditContainer"
>
  <h3>
    CREATE A SUBREDDIT
  </h3>
  <div />
  <hr />
  <form
    onSubmit={[Function]}
  >
    <div
      className="formGroupSubredditComponent"
    >
      <label
        for="SubredditName"
      >
        Enter Subreddit Name
      </label>
      <br />
      <input
        id="srSubredditName"
        name="text"
        onChange={[Function]}
        placeholder="Think of something remarkable!"
        type="text"
      />
    </div>
    <div
      className="formGroupSubredditComponent"
    >
      <label
        for="SubredditRule"
      >
        Subreddit Rule:
      </label>
      <textarea
        id="srSubredditRule1"
        name="text"
        onChange={[Function]}
        placeholder="ex: Flamers will be banned! "
        type="textarea"
      />
    </div>
    <div
      className="formGroupSubredditComponent"
    >
      <label
        for="SubredditRule"
      >
        Subreddit Rule:
      </label>
      <textarea
        id="srSubredditRule2"
        name="text"
        onChange={[Function]}
        placeholder="ex: Flamers will be banned! "
        type="textarea"
      />
    </div>
    <div
      className="formGroupSubredditComponent"
    >
      <label
        for="SubredditRule"
      >
        Subreddit Rule:
      </label>
      <textarea
        id="srSubredditRule3"
        name="text"
        onChange={[Function]}
        placeholder="ex: Flamers will be banned! "
        type="textarea"
      />
    </div>
    <div
      className="formGroupSubredditComponent"
    >
      <label
        for="SubredditBio"
      >
        Subreddit Bio:
      </label>
      <textarea
        id="srSubredditBio"
        name="text"
        onChange={[Function]}
        placeholder="ex: This subreddit is for people who..etc. "
        type="textarea"
      />
    </div>
    <div
      className="formGroupSubredditComponent"
    >
      <label
        for="SubredditModeator"
      >
        Subreddit Moderator:
      </label>
      <textarea
        id="srSubredditModerator"
        name="text"
        onChange={[Function]}
        placeholder="Usernames separated by commas. EX: Mahmoud, Mohamed, JamesBond"
        type="textarea"
      />
      <label
        for="coverPhoto"
      >
        Cover Photo:
      </label>
      <input
        id="coverPhoto"
        name="coverPhoto"
        onChange={[Function]}
        type="file"
      />
    </div>
    <button
      className="srSubredditPageCreateButton"
    >
      CREATE
    </button>
  </form>
</div>
`;

exports[`CreateSubreddit component snapshot it matches the snapshot 1`] = `
<div
  className="createSubredditContainer"
>
  <h3>
    CREATE A SUBREDDIT
  </h3>
  <div />
  <hr />
  <form
    onSubmit={[Function]}
  >
    <div
      className="formGroupSubredditComponent"
    >
      <label
        for="SubredditName"
      >
        Enter Subreddit Name
      </label>
      <br />
      <input
        id="srSubredditName"
        name="text"
        onChange={[Function]}
        placeholder="Think of something remarkable!"
        type="text"
      />
    </div>
    <div
      className="formGroupSubredditComponent"
    >
      <label
        for="SubredditRule"
      >
        Subreddit Rule:
      </label>
      <textarea
        id="srSubredditRule1"
        name="text"
        onChange={[Function]}
        placeholder="ex: Flamers will be banned! "
        type="textarea"
      />
    </div>
    <div
      className="formGroupSubredditComponent"
    >
      <label
        for="SubredditRule"
      >
        Subreddit Rule:
      </label>
      <textarea
        id="srSubredditRule2"
        name="text"
        onChange={[Function]}
        placeholder="ex: Flamers will be banned! "
        type="textarea"
      />
    </div>
    <div
      className="formGroupSubredditComponent"
    >
      <label
        for="SubredditRule"
      >
        Subreddit Rule:
      </label>
      <textarea
        id="srSubredditRule3"
        name="text"
        onChange={[Function]}
        placeholder="ex: Flamers will be banned! "
        type="textarea"
      />
    </div>
    <div
      className="formGroupSubredditComponent"
    >
      <label
        for="SubredditBio"
      >
        Subreddit Bio:
      </label>
      <textarea
        id="srSubredditBio"
        name="text"
        onChange={[Function]}
        placeholder="ex: This subreddit is for people who..etc. "
        type="textarea"
      />
    </div>
    <div
      className="formGroupSubredditComponent"
    >
      <label
        for="SubredditModeator"
      >
        Subreddit Moderator:
      </label>
      <textarea
        id="srSubredditModerator"
        name="text"
        onChange={[Function]}
        placeholder="Usernames separated by commas. EX: Mahmoud, Mohamed, JamesBond"
        type="textarea"
      />
      <label
        for="coverPhoto"
      >
        Cover Photo:
      </label>
      <input
        id="coverPhoto"
        name="coverPhoto"
        onChange={[Function]}
        type="file"
      />
    </div>
    <button
      className="srSubredditPageCreateButton"
    >
      CREATE
    </button>
  </form>
</div>
`;
